dbms/src/Storages/Page/V3/Universal/UniversalPageIdFormat.h

```diff
@@ -31,8 +31,14 @@ namespace DB
 //
 //
 // The main key format types:
-// Raft related key
+// Raft related key format
 //  Format: https://github.com/tikv/tikv/blob/9c0df6d68c72d30021b36d24275fdceca9864235/components/keys/src/lib.rs#L24
+//  And because some key will be migrated from kv engine to raft engine,
+//  kv engine and raft engine may write and delete the same key.
+//  So to distinguish data written by kv engine and raft engine, we prepend an `0x01` to the key written by kv engine.
+//  For example, suppose a key in tikv to be {0x01, 0x02, 0x03}.
+//  If it is written by raft engine, then actual key uni ps see is the same as in tikv.
+//  But if it is written by kv engine, the actual key uni ps see will be {0x01, 0x01, 0x02, 0x03}.
```

Comment:
Will this cause problems with scanning raft keys? If raft-engine scan with '\x01' prefix, it will unexpectedly see some keys actually written to the kv engine.

Reply:
what about this?

- prepend another '\x01' for raft engine keys
- prepend '\x02' for kv engine keys

Reply:
Then we should let the tikv's raft-engine project be aware of '\x02' prefix is reserved, and not to use this prefix in the future. /cc @CalvinNeo

Reply:
This should not be difficult. The convention that all raft keys in tikv start with `0x01` won't easily be changed.